### PR TITLE
Fix ONNX export for dynamic batch dimensions

### PR DESCRIPTION
**Pull request by svengoluza**
## What does this PR do?

ONNX export silently baked spatial dimensions and batch size into the graph as fixed constants. When users tried to run the exported model with a different batch size at inference time, it would fail because the ONNX graph contained hardcoded values derived from the export-time batch size.

Three root causes:

1. **`LayerNorm.forward` used `x.size(3)` instead of `self.normalized_shape`** — The ONNX tracer captured the concrete integer from the export-time tensor shape and embedded it as a constant node. Switching to `self.normalized_shape` (already stored on the module) gives the tracer a static attribute it can reference symbolically.

2. **`spatial_shapes` was built as a Python list, then converted with `torch.as_tensor()`** — The tracer never saw the `h, w` values flow through tensor operations, so it treated them as constants. Building `spatial_shapes` directly as a tensor with index assignment (`spatial_shapes[lvl, 0] = h`) lets the tracer track the symbolic relationship between the input spatial dims and downstream uses.

3. **`gen_encoder_output_proposals` created `valid_H`/`valid_W` via Python list comprehensions** — `torch.tensor([H_ for _ in range(N_)])` bakes the batch-dependent value `N_` as a fixed list length. Replacing this with `H_.expand(N_)` keeps the batch dimension symbolic in the graph.

Additionally, a `dynamic_batch` parameter is added to `RFDETR.export_onnx()` and the standalone export CLI. When enabled, it marks the batch axis (`dim 0`) as dynamic on all input and output names, allowing the exported model to accept variable batch sizes at runtime.

**Related Issue(s):** https://github.com/roboflow/rf-detr/issues/376 , https://github.com/roboflow/rf-detr/pull/79

## Type of Change

- Bug fix (non-breaking change that fixes an issue)

## Testing

- [x] I have tested this change locally

### CODE REVIEW

File: src/rfdetr/models/transformer.py [hunk 1]
```diff
@@ -224,12 +224,22 @@ def forward(self, srcs, masks, pos_embeds, refpoint_embed, query_feat):
         src_flatten = []
         mask_flatten = [] if masks is not None else None
         lvl_pos_embed_flatten = []
-        spatial_shapes = []
+        # Build spatial_shapes as a tensor directly so that the ONNX tracer
+        # can track h/w symbolically instead of baking them in as constants.
+        spatial_shapes = torch.empty(
+            (len(srcs), 2),
+            device=srcs[0].device,
+            dtype=torch.long,
```
Comment (Borda) [inline src/rfdetr/models/transformer.py]:
```suggestion
            dtype=torch.long
```

**Review by Borda — APPROVED** [review]
